soc/nordic/common/CMakeLists.txt

```diff
@@ -12,6 +12,7 @@ if(CONFIG_ARM)
 endif()
 
 zephyr_library_sources_ifdef(CONFIG_POWEROFF poweroff.c)
+zephyr_library_sources_ifdef(CONFIG_CPU_HAS_CUSTOM_FIXED_SOC_MPU_REGIONS mpu_regions.c)
```

Comment:
This seems applicable only to selected SoCs. How about renaming the file to something like nrf54h_nrf92_mpu_regions.c and adding some more configuration guards to make sure it's not included in nRF54L or nRF91 builds?

Reply:
When we resolve this comment, I'll approve.

Reply:
Alright, I have renamed it to `nrf54hx_nrf92x_mpu_regions.c` and update the `if` condition to check for the appropriate SoC series.